### PR TITLE
[C++][Qt5] Fixes segmentation faults for arrays of primitive types

### PR DESCRIPTION
**Pull request by etherealjoy**
(cherry picked from commit bfaf253caebe0a8367275aa477d9bc0dcd030adc)

### PR checklist

- [X] Read the [contribution guidelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [X] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [X] Filed the PR against the correct branch: `3.0.0` branch for changes related to OpenAPI spec 3.0. Default: `master`.
- [X] Copied the [technical committee](https://github.com/swagger-api/swagger-codegen/#swagger-codegen-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

(details of the change, additional tests that have been done, reference to the issue for tracking, etc)
Fixes for #6984

### CODE REVIEW

**Comment by etherealjoy** [comment]
@ravinikam @stkrwork @fvarose
Adding Technical committee to review changes

File: modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache [hunk 1]
```diff
@@ -132,64 +132,113 @@ setValue(void* value, QJsonValue obj, QString type, QString complexType) {
...
-                    setValue(&val, jval, QStringLiteral("QDate"), QStringLiteral(""));
-                    output->append((void*)val);
-                }
-                else if(QStringLiteral("QDateTime").compare(complexType) == 0) {
-                    QDateTime * val = new QDateTime();
-                    setValue(&val, jval, QStringLiteral("QDateTime"), QStringLiteral(""));
-                    output->append((void*)val);
-                }
+            QList<SWGObject*> **val = static_cast<QList<SWGObject*>**>(value);
+            foreach (auto item, **val) {
```
Comment (stkrwork) [inline modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache]: why not use for (auto item : **val)

File: modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache [hunk 2]
```diff
@@ -132,64 +132,113 @@ setValue(void* value, QJsonValue obj, QString type, QString complexType) {
...
+                delete item;
+            }
+            delete *val;
+            *val = output;
+        }
+        else if(QStringLiteral("qint32").compare(complexType) == 0) {
+            QList<qint32> **output = reinterpret_cast<QList<qint32> **> (value);
+            (*output)->clear();
+            QJsonArray arr = obj.toArray();
+            foreach (const QJsonValue & jval, arr){
```
Comment (stkrwork) [inline modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache]: similar as above

File: modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache [hunk 3]
```diff
@@ -132,64 +132,113 @@ setValue(void* value, QJsonValue obj, QString type, QString complexType) {
...
+                qint32 val;
+                setValue(&val, jval, QStringLiteral("qint32"), QStringLiteral(""));
+                (*output)->push_back(val);
+            }
+        }
+        else if(QStringLiteral("qint64").compare(complexType) == 0) {
+            QList<qint64> **output = reinterpret_cast<QList<qint64> **> (value);
+            (*output)->clear();
+            QJsonArray arr = obj.toArray();
+            foreach (const QJsonValue & jval, arr){
```
Comment (stkrwork) [inline modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache]: similar as above

File: modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache [hunk 4]
```diff
@@ -132,64 +132,113 @@ setValue(void* value, QJsonValue obj, QString type, QString complexType) {
...
+                qint64 val;
+                setValue(&val, jval, QStringLiteral("qint64"), QStringLiteral(""));
+                (*output)->push_back(val);
+            }
+        }
+        else if(QStringLiteral("bool").compare(complexType) == 0) {
+            QList<bool> **output = reinterpret_cast<QList<bool> **> (value);
+            (*output)->clear();
+            QJsonArray arr = obj.toArray();
+            foreach (const QJsonValue & jval, arr){
```
Comment (stkrwork) [inline modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache]: similar as above

File: modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache [hunk 5]
```diff
@@ -132,64 +132,113 @@ setValue(void* value, QJsonValue obj, QString type, QString complexType) {
...
+                bool val;
+                setValue(&val, jval, QStringLiteral("bool"), QStringLiteral(""));
+                (*output)->push_back(val);
+            }
+        }
+        else if(QStringLiteral("float").compare(complexType) == 0) {
+            QList<float> **output = reinterpret_cast<QList<float> **> (value);
+            (*output)->clear();
+            QJsonArray arr = obj.toArray();
+            foreach (const QJsonValue & jval, arr){
```
Comment (stkrwork) [inline modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache]: similar as above

File: modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache [hunk 6]
```diff
@@ -132,64 +132,113 @@ setValue(void* value, QJsonValue obj, QString type, QString complexType) {
...
+                float val;
+                setValue(&val, jval, QStringLiteral("float"), QStringLiteral(""));
+                (*output)->push_back(val);
+            }
+        }
+        else if(QStringLiteral("double").compare(complexType) == 0) {
+            QList<double> **output = reinterpret_cast<QList<double> **> (value);
+            (*output)->clear();
+            QJsonArray arr = obj.toArray();
+            foreach (const QJsonValue & jval, arr){
```
Comment (stkrwork) [inline modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache]: similar as above

File: modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache [hunk 7]
```diff
@@ -132,64 +132,113 @@ setValue(void* value, QJsonValue obj, QString type, QString complexType) {
...
+                double val;
+                setValue(&val, jval, QStringLiteral("double"), QStringLiteral(""));
+                (*output)->push_back(val);
+            }
+        }
+        else if(QStringLiteral("QString").compare(complexType) == 0) {
+            QList<QString*> **output = reinterpret_cast<QList<QString*> **> (value);
+            foreach (auto item, **output) {
```
Comment (stkrwork) [inline modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache]: similar as above

File: modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache [hunk 8]
```diff
@@ -132,64 +132,113 @@ setValue(void* value, QJsonValue obj, QString type, QString complexType) {
...
+                delete item;
+            }
+            (*output)->clear();
+            QJsonArray arr = obj.toArray();
+            foreach (const QJsonValue & jval, arr){
```
Comment (stkrwork) [inline modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache]: similar as above

File: modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache [hunk 9]
```diff
@@ -132,64 +132,113 @@ setValue(void* value, QJsonValue obj, QString type, QString complexType) {
...
+                QString * val = new QString();
+                setValue(&val, jval, QStringLiteral("QString"), QStringLiteral(""));
+                (*output)->push_back(val);
+            }
+        }
+        else if(QStringLiteral("QDate").compare(complexType) == 0) {
+            QList<QDate*> **output = reinterpret_cast<QList<QDate*> **> (value);
+            foreach (auto item, **output) {
```
Comment (stkrwork) [inline modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache]: similar as above

File: modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache [hunk 10]
```diff
@@ -132,64 +132,113 @@ setValue(void* value, QJsonValue obj, QString type, QString complexType) {
...
+                delete item;
+            }
+            (*output)->clear();
+            QJsonArray arr = obj.toArray();
+            foreach (const QJsonValue & jval, arr){
```
Comment (stkrwork) [inline modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache]: similar as above

File: modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache [hunk 11]
```diff
@@ -132,64 +132,113 @@ setValue(void* value, QJsonValue obj, QString type, QString complexType) {
...
+                QDate * val = new QDate();
+                setValue(&val, jval, QStringLiteral("QDate"), QStringLiteral(""));
+                (*output)->push_back(val);
+            }
+        }
+        else if(QStringLiteral("QDateTime").compare(complexType) == 0) {
+            QList<QDateTime*> **output = reinterpret_cast<QList<QDateTime*> **> (value);
+            foreach (auto item, **output) {
```
Comment (stkrwork) [inline modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache]: similar as above

File: modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache [hunk 12]
```diff
@@ -132,64 +132,113 @@ setValue(void* value, QJsonValue obj, QString type, QString complexType) {
...
+                delete item;
+            }
+            (*output)->clear();
+            QJsonArray arr = obj.toArray();
+            foreach (const QJsonValue & jval, arr){
```
Comment (stkrwork) [inline modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache]: similar as above

File: modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache [hunk 13]
```diff
@@ -254,11 +303,52 @@ toJsonValue(QString name, void* value, QJsonObject* output, QString type) {
 
 void
 toJsonArray(QList<void*>* value, QJsonArray* output, QString innerName, QString innerType) {
-    foreach(void* obj, *value) {
-        QJsonObject element;
-
-        toJsonValue(nullptr, obj, &element, innerType);
-        output->append(element);
+    if(innerType.startsWith("SWG")){
+        foreach(void* obj, *value) {
```
Comment (stkrwork) [inline modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache]: similar as above

File: modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache [hunk 14]
```diff
@@ -254,11 +303,52 @@ toJsonValue(QString name, void* value, QJsonObject* output, QString type) {
...
+            SWGObject *swgObject = reinterpret_cast<SWGObject *>(obj);
+            if(swgObject != nullptr) {
+                output->append(*(swgObject->asJsonObject()));
+            }
+        }
+    }
+    else if(QStringLiteral("QString").compare(innerType) == 0) {
+        foreach(QString* obj, *(reinterpret_cast<QList<QString*>*>(value))){
```
Comment (stkrwork) [inline modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache]: similar as above

File: modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache [hunk 15]
```diff
@@ -254,11 +303,52 @@ toJsonValue(QString name, void* value, QJsonObject* output, QString type) {
...
+            output->append(QJsonValue(*obj));
+        }
+    }
+    else if(QStringLiteral("QDate").compare(innerType) == 0) {
+        foreach(QDate* obj, *(reinterpret_cast<QList<QDate*>*>(value))){
```
Comment (stkrwork) [inline modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache]: similar as above

File: modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache [hunk 16]
```diff
@@ -254,11 +303,52 @@ toJsonValue(QString name, void* value, QJsonObject* output, QString type) {
...
+            output->append(QJsonValue(obj->toString(Qt::ISODate)));
+        }
+    }
+    else if(QStringLiteral("QDateTime").compare(innerType) == 0) {
+        foreach(QDateTime* obj, *(reinterpret_cast<QList<QDateTime*>*>(value))){
```
Comment (stkrwork) [inline modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache]: similar as above

File: modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache [hunk 17]
```diff
@@ -254,11 +303,52 @@ toJsonValue(QString name, void* value, QJsonObject* output, QString type) {
...
+            output->append(QJsonValue(obj->toString(Qt::ISODate)));        }
+    }
+    else if(QStringLiteral("QByteArray").compare(innerType) == 0) {
+        foreach(QByteArray* obj, *(reinterpret_cast<QList<QByteArray*>*>(value))){
```
Comment (stkrwork) [inline modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache]: similar as above

File: modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache [hunk 18]
```diff
@@ -254,11 +303,52 @@ toJsonValue(QString name, void* value, QJsonObject* output, QString type) {
...
+            output->append(QJsonValue(QString(obj->toBase64())));
+        }
+    }
+    else if(QStringLiteral("qint32").compare(innerType) == 0) {
+        foreach(qint32 obj, *(reinterpret_cast<QList<qint32>*>(value)))
```
Comment (stkrwork) [inline modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache]: similar as above

File: modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache [hunk 19]
```diff
@@ -254,11 +303,52 @@ toJsonValue(QString name, void* value, QJsonObject* output, QString type) {
...
+            output->append(QJsonValue(obj));
+    }
+    else if(QStringLiteral("qint64").compare(innerType) == 0) {
+        foreach(qint64 obj, *(reinterpret_cast<QList<qint64>*>(value)))
```
Comment (stkrwork) [inline modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache]: similar as above

File: modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache [hunk 20]
```diff
@@ -254,11 +303,52 @@ toJsonValue(QString name, void* value, QJsonObject* output, QString type) {
...
+            output->append(QJsonValue(obj));
+    }
+    else if(QStringLiteral("bool").compare(innerType) == 0) {
+        foreach(bool obj, *(reinterpret_cast<QList<bool>*>(value)))
```
Comment (stkrwork) [inline modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache]: similar as above

File: modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache [hunk 21]
```diff
@@ -254,11 +303,52 @@ toJsonValue(QString name, void* value, QJsonObject* output, QString type) {
...
+            output->append(QJsonValue(obj));
+    }
+    else if(QStringLiteral("float").compare(innerType) == 0) {
+        foreach(float obj, *(reinterpret_cast<QList<float>*>(value)))
```
Comment (stkrwork) [inline modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache]: similar as above

File: modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache [hunk 22]
```diff
@@ -254,11 +303,52 @@ toJsonValue(QString name, void* value, QJsonObject* output, QString type) {
...
+            output->append(QJsonValue(obj));
+    }
+    else if(QStringLiteral("double").compare(innerType) == 0) {
+        foreach(double obj, *(reinterpret_cast<QList<double>*>(value)))
```
Comment (stkrwork) [inline modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache]: similar as above

File: modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache [hunk 23]
```diff
@@ -132,64 +132,113 @@ setValue(void* value, QJsonValue obj, QString type, QString complexType) {
...
     else if(type.startsWith("QList") && QString("").compare(complexType) != 0 && obj.isArray()) {
         // list of values
-        QList<void*>* output = new QList<void*>();
-        QJsonArray arr = obj.toArray();
-        foreach (const QJsonValue & jval, arr) {
-            if(complexType.startsWith("SWG")) {
+        if(complexType.startsWith("SWG")) {
+            QList<SWGObject*>* output = new QList<SWGObject*>();
+            QJsonArray arr = obj.toArray();
+            foreach (const QJsonValue & jval, arr) {
```
Comment (stkrwork) [inline modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache]: why not use for (const QJsonValue & jval : arr)

File: modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache (hunk 22 above)
Comment (etherealjoy) [reply modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache]: Done

File: modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache (hunk 21 above)
Comment (etherealjoy) [reply modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache]: Done

File: modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache (hunk 20 above)
Comment (etherealjoy) [reply modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache]: Done

File: modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache (hunk 19 above)
Comment (etherealjoy) [reply modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache]: Done

File: modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache (hunk 18 above)
Comment (etherealjoy) [reply modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache]: Done

File: modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache (hunk 17 above)
Comment (etherealjoy) [reply modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache]: Done

File: modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache (hunk 16 above)
Comment (etherealjoy) [reply modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache]: Done

File: modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache (hunk 15 above)
Comment (etherealjoy) [reply modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache]: Done

File: modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache (hunk 14 above)
Comment (etherealjoy) [reply modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache]: Done

File: modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache (hunk 13 above)
Comment (etherealjoy) [reply modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache]: Done

File: modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache (hunk 12 above)
Comment (etherealjoy) [reply modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache]: Done

File: modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache (hunk 11 above)
Comment (etherealjoy) [reply modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache]: Done

File: modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache (hunk 10 above)
Comment (etherealjoy) [reply modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache]: Done

File: modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache (hunk 9 above)
Comment (etherealjoy) [reply modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache]: Done

File: modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache (hunk 8 above)
Comment (etherealjoy) [reply modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache]: Done

File: modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache (hunk 7 above)
Comment (etherealjoy) [reply modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache]: Done

File: modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache (hunk 6 above)
Comment (etherealjoy) [reply modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache]: Done

File: modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache (hunk 5 above)
Comment (etherealjoy) [reply modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache]: Done

File: modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache (hunk 4 above)
Comment (etherealjoy) [reply modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache]: Done

File: modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache (hunk 3 above)
Comment (etherealjoy) [reply modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache]: Done

File: modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache (hunk 2 above)
Comment (etherealjoy) [reply modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache]: Done

File: modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache (hunk 1 above)
Comment (etherealjoy) [reply modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache]: Done

File: modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache (hunk 23 above)
Comment (etherealjoy) [reply modules/swagger-codegen/src/main/resources/qt5cpp/helpers-body.mustache]: Done

**Comment by wing328** [comment]
Sorry, I missed the review and approval by @stkrwork 

Just merged this PR into master. Thanks for the contribution.

**Pull request merged**